### ISSUE TITLE
[cmake] warns it can't find libedit

### ISSUE THREAD

**Issue by LW-archlinux** [llvm/llvm-project#62300]
Since commit https://github.com/llvm/llvm-project/commit/7d89e6c1c0a01c959beddab643e39d2bc4c2c97d I get the following warning when building my https://aur.archlinux.org/packages/libclc-minimal-git package.

```console
CMake Warning at /usr/lib/cmake/llvm/LLVMConfig.cmake:167 (find_package):
  By not providing "FindLibEdit.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "LibEdit", but
  CMake did not find one.

  Could not find a package configuration file provided by "LibEdit" with any
  of the following names:

    LibEditConfig.cmake
    libedit-config.cmake

  Add the installation prefix of "LibEdit" to CMAKE_PREFIX_PATH or set
  "LibEdit_DIR" to a directory containing one of the above files.  If
  "LibEdit" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  CMakeLists.txt:46 (find_package)
```

Reverting the commit doesn't show the warning and there's no error about a missing target.

libedit is installed with pkgconf `.pc` files but no CMake files for it are present. As far as I can tell the warning is not relevant for the libclc-minimal-git build.

Maybe the warning is not applicable for all projects?

@ekilmer, please describe what were the circumstances where you noticed 

> CMake will report an error about a missing target.

as the description of 7d89e6c1c0a01c959beddab643e39d2bc4c2c97d states.

Lone_Wolf

**Comment by ekilmer** [comment]
The message saying

> By not providing "FindLibEdit.cmake" in CMAKE_MODULE_PATH

means that `FindLibEdit.cmake` located at [`cmake/Modules/FindLibEdit.cmake`](https://github.com/llvm/llvm-project/blob/104448f9c5d2f82e6c5e09270ec540bdbee84d47/cmake/Modules/FindLibEdit.cmake) is likely not installed, which produces the warning.

> Maybe the warning is not applicable for all projects ?

Correct. Unless you are linking against LLVM's `LLVMLineEditor` target, you don't _need_ to find libedit, but you will see a warning if CMake cannot find a `FindLibEdit.cmake`.

----------

As for me, I received an error trying to build Clang as a standalone project after pointing it to an installation of LLVM.

Reproduction steps for my error:

* Ensure libedit is installed, `libedit-devel` package on Fedora, my dev environment
* Check out `llvm-project` repo and navigate to it (I used commit 104448f9c5d2f82e6c5e09270ec540bdbee84d47)
* Revert https://github.com/llvm/llvm-project/commit/7d89e6c1c0a01c959beddab643e39d2bc4c2c97d (so that an error will appear)

```bash
$ cmake -B build-llvm -S llvm -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_INSTALL_PREFIX=$(pwd)/install-llvm \
    -G Ninja

# Ensure libedit is found and used before building
$ cmake --build build-llvm

$ cmake --build build-llvm --target install

$ cmake -B build-clang -S clang -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_INSTALL_PREFIX=$(pwd)/install-clang \
    -DLLVM_INCLUDE_TESTS=OFF \
    -DLLVM_DIR=$(pwd)/install-llvm/lib/cmake/llvm \
    -G Ninja
[...]
CMake Error at $PWD/install-llvm/lib/cmake/llvm/LLVMExports.cmake:1222 (set_target_properties):
  The link interface of target "LLVMLineEditor" contains:

    LibEdit::LibEdit

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
```

Granted, while my just-built installation of LLVM (and anyone else who builds directly from source) does not install `FindLibEdit.cmake`, the standalone Clang build adds the root `cmake/modules` path to `CMAKE_MODULE_PATH` so that it can find `FindLibEdit.cmake`.

I also noticed that Fedora packages it in their `llvm-devel` package so that other packages can find that file:

```sh
$ dnf provides "*FindLibEdit.cmake"
llvm-devel-15.0.7-1.fc37.x86_64 : Libraries and header files for LLVM
Repo        : @System
Matched from:
Other       : *FindLibEdit.cmake
```

I'm not very familiar with packaging for Fedora, but I believe the relevant lines where Fedora installs the common CMake modules is here https://src.fedoraproject.org/rpms/llvm/blob/072d2198fb221a44f1c03e9d308fe45bf988ab2e/f/llvm.spec#_425

**Comment by ekilmer** [comment]
Reading more of the [`cmake/README.rst`](https://github.com/llvm/llvm-project/blob/main/cmake/README.rst)

> Modules that are reachable from installed modules should instead go in `${project}/cmake/modules` of the most upstream project that uses them.

So, in my commit, I think I may have missed moving `FindLibEdit.cmake` to LLVM's CMake module directory and installing it.

LLDB project also seems to depend on LibEdit
```
$ git grep LibEdit::LibEdit
[...]
  lldb/source/Host/CMakeLists.txt:146:  list(APPEND EXTRA_LIBS LibEdit::LibEdit)
  lldb/source/Host/CMakeLists.txt:156:  list(APPEND LLDB_LIBEDIT_LIBS LibEdit::LibEdit)
  lldb/source/Plugins/ScriptInterpreter/Python/CMakeLists.txt:13:  list(APPEND LLDB_LIBEDIT_LIBS LibEdit::LibEdit)
[...]
```

**Comment by ekilmer** [comment]
Important note: https://github.com/llvm/llvm-project/commit/7d89e6c1c0a01c959beddab643e39d2bc4c2c97d didn't necessarily fix the workflow to build clang standalone, as described in https://github.com/llvm/llvm-project/issues/62300#issuecomment-1518665525---I still need to add `-DCMAKE_MODULE_PATH=$(pwd)/cmake/Modules` when building Clang to get it to compile.

**Comment by ekilmer** [comment]
@LW-archlinux I've opened https://reviews.llvm.org/D148993, which should install `FindLibEdit.cmake` with LLVM by default now.
